src/sql/azdata.d.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'azdata' {
+	import * as vscode from 'vscode';
+
+	/**
+	 * Connection profile primary class
+	 */
+	export class ConnectionProfile {
+		providerId: string;
+		connectionId: string;
+		connectionName: string;
+		serverName: string;
+		databaseName: string;
+		userName: string;
+		password: string;
+		authenticationType: string;
+		savePassword: boolean;
+		groupFullName: string;
+		groupId: string;
+		saveProfile: boolean;
+		azureTenantId?: string;
+
```

Comment:
This class doesn't have a `options: Map<string, any>` member, so we need to decide how to handle properties that aren't listed above.  We can either add an `options` property, or just allow the createFrom to add the addtional properties to the object anyways.  Then consumers can downcast to a different interface based on the providerId and access the "unknown" properties.